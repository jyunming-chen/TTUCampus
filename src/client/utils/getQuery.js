// @ts-check

/**
 * @param {string} querystring e.g. `?aaa=111&bbb=222`
 * @return {Partial<Record<string, string>>}
 */
export default function getQuery(querystring) {
  return decodeURI(querystring)
    .slice(1) // remove `?`
    .split('&')
    .filter(x => x.length > 0) // remove ''
    .reduce((query, segment) => {
      const splitAt = segment.indexOf('=');
      const key = splitAt === -1 ? segment : segment.slice(0, splitAt);
      const value = splitAt === -1 ? undefined : segment.slice(splitAt + 1);
      return Object.assign({}, query, { [key]: value });
    }, {});
}
