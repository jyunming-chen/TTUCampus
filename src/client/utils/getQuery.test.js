import getQuery from './getQuery';

test('getQuery("?a=1&b=2")', () => {
  expect(getQuery('?a=1&b=2')).toMatchSnapshot();
});

test('getQuery("")', () => {
  expect(getQuery('')).toMatchSnapshot();
});

test('getQuery("?a&b=1")', () => {
  expect(getQuery('?a&b=1')).toMatchSnapshot();
});
