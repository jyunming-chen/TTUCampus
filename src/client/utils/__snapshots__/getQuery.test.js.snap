// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getQuery("") 1`] = `Object {}`;

exports[`getQuery("?a&b=1") 1`] = `
Object {
  "a": undefined,
  "b": "1",
}
`;

exports[`getQuery("?a=1&b=2") 1`] = `
Object {
  "a": "1",
  "b": "2",
}
`;
